packages/dart_firebase_admin/lib/src/google_cloud_firestore/firestore.dart

```diff
@@ -193,6 +196,79 @@ class Firestore {
 
     return reader.get(tag);
   }
+
+  Future<T> runTransaction<T>(
+    TransactionHandler<T> transactionHandler, {
+    int maxAttempts = 5,
+    Duration timeout = const Duration(seconds: 30),
+  }) async {
+    if (maxAttempts <= 0) {
+      return Future.error(
+        FirebaseFirestoreAdminException(
+          FirestoreClientErrorCode.failedPrecondition,
+          'Transaction max attemps must be > 0.',
+        ),
+      );
+    }
+    return Future<T>(() async {
+      var attempts = 1;
+      while (true) {
```

Comment:
I think the following would be a more defensive style: 
```
var attempts = 0;
while(++attempts <= maxAttempts) {
```
and starting line 222:
```
if (attempts >= maxAttempts) {
  return Future.error(...);
}
```